docs/writing-easyconfig-files.md

```diff
@@ -940,6 +940,70 @@ Example:
+   Check the toolchain easyconfig file (e.g. `foss-2023b.eb`) to find the versions of the sub-toolchain.
+   For very recent toolchains you can also check the `develop` branch of the [easyconfigs git repository](https://github.com/easybuilders/easybuild-easyconfigs) and open PRs.
+1. If an easyconfig for the listed dependency with the new toolchain already exists use its version in your new easyconfig.
+   Otherwise, you need to create a new easyconfig for the listed dependency with the new toolchain version (based on that of another toolchain version).
+   The steps are the same and also need to be repeated for each dependency of this dependency.
+1. Finally, build and test your easyconfig and consider [contributing](contributing.md#contributing-easyconfig-files--contributing_easyconfigs) it.
+
+Especially when creating easyconfigs for dependencies it is a good idea to put all of them into a new folder.
+To allow EasyBuild to find these easyconfigs in your folder, pass its path via `eb --robot <path> ...`.
+
+This manual process can be partially automated:
+
+1. Create a new folder for the new easyconfig and its dependencies.
+1. Run `easybuild <current_easyconfig.eb> --try-toolchain-version=<new-version> --try-update-deps --copy-ec <folder>`.
+   This will update the toolchain version, search for the versions of dependencies in the new toolchain and use them if available.
+   If a dependency with the new toolchain is not available this will create a new easyconfig for this toolchain
+   based on an existing one for another toolchain (version).
+   All newly create easyconfigs will be copied to the specified folder.
+1. Optionally, but recommended, update the version of the software for each created easyconfig to the latest available one.
+1. **Important**: Verify the newly created easyconfigs by comparing each of them against the existing one for that software.
+   Only the version of the software, toolchain and dependencies should have been changed.
+   EasyBuild often replaces local variables or templates by their value when updating easyconfigs.
+   It might be easier to use those easyconfigs just for getting the versions of each dependency required
+   and still copy & update the easyconfigs from existing ones manually.
+
+When updating a software version the patches in the easyconfig need to be checked too.
+If they don't apply they might not be required anymore, when the upstream software fixed the issue, and can be removed.
+Otherwise, they need to be adjusted for the changes in the source code.
+
+For updates to easyconfigs containing Python packages (`PythonBundle`) you should check each of them if there is a newer version and if it is still required.
+The latest version of a single Python package can usually be found on [PyPI](https://pypi.org).
+To find the list of Python dependencies for another Python package to be installed by an easyconfig you can use the `findPythonDeps` script
```

Comment:
Perhaps add more info on actually run `findPythonDeps`?

I got the following from @ocaisa which was very useful in my case :

`$(dirname $(which eb))/../easybuild/scripts/findPythonDeps.py --ec snakemake-8.4.2-foss-2024a.eb snakemake==8.27.0`

Reply:
Sorry. I often find documentation hard to follow. (Typically examples on Slack work much better for me.) I searched for `findPythonDeps` and `EasyBuild` on google, and the took me to 

https://docs.easybuild.io/api/easybuild/scripts/findPythonDeps/

but from the sparse documentation on that page, there is no way I would be able to come up with `$(dirname $(which eb))/../easybuild/scripts/findPythonDeps.py --ec snakemake-8.4.2-foss-2024a.eb snakemake==8.27.0`. I mean I wouldn't even know where to look for the script `findPythonDeps.py` as it is not on the `PATH` (at least not on my system).

Reply:
I agree that the script is not easy to find (it also took me a while yesterday). In an ideal world we'd have an environment variable that defined the location of the scripts, but that doesn't cover the scenario where someone has installed EasyBuild via pip...what I gave was the best I could come up with that _might_ be universal

Reply:
I see. Added that and opened another PR to improve that docs site